ee/modules/030-cloud-provider-openstack/hooks/discover.go

```diff
@@ -0,0 +1,198 @@
+/*
+Copyright 2023 Flant JSC
+Licensed under the Deckhouse Platform Enterprise Edition (EE) license. See https://github.com/deckhouse/deckhouse/blob/main/ee/LICENSE
+*/
+
+package hooks
+
+import (
+	"fmt"
+	"regexp"
+	"sort"
+	"strings"
+	"unicode"
+
+	"github.com/flant/addon-operator/pkg/module_manager/go_hook"
+	"github.com/flant/addon-operator/sdk"
+	"github.com/flant/shell-operator/pkg/kube_events_manager/types"
+	log "github.com/sirupsen/logrus"
```

Comment:
do not use `logrus` in hooks. Please use `input.LogEntry` as [here](https://github.com/deckhouse/deckhouse/blob/468ae94c1d944d253b2d62a5e26050d1efa0a823/modules/040-control-plane-manager/hooks/etcd_quota_backend_bytes.go#L240C4-L240C18)